app/controllers/submissions_controller.rb

```diff
@@ -1,6 +1,17 @@
 class SubmissionsController < ApplicationController
-  skip_before_action :authenticate_user!, only: [:new, :create, :thank_you]
-  layout 'dashboard', only: [:valid, :rated, :to_rate, :rejected, :results]
+  skip_before_action :authenticate_user!, only: [:confirm, :new, :create, :thank_you]
+  layout 'dashboard', only: [:valid, :all, :rated, :to_rate, :rejected, :results]
+
+  def confirm
+    token = params[:confirmation_token]
+    submission = Submission.find_by(confirmation_token: token)
+    if submission.has_expired?
```

Comment:
or you could raise exception `Submission.find_by!`

using `if submission.try(&:has_expired?)` will trigger `else` and `submission.confirmed!` will be called on `nil`